src/Service/EncryptionService.php

```diff
@@ -56,9 +75,15 @@ public function encryptField($value): EncryptedColumnVO
     {
         if ($value instanceof LazyLoadingInterface) {
             /** @var LazyLoadingInterface|ValueHolderInterface $value */
+            // If the value hasn't been decrypted; it hasn't been changed. Don't bother reencrypting unless it
+            // was encrypted using a different configuration
             if (!$value->isProxyInitialized()) {
-                //if data hasn't been encrypted, we don't need to change it; set it back to what it was at load
-                return $this->originalValues[spl_object_hash($value)];
+                $original = $this->originalValues[spl_object_hash($value)];
+                if (
+                    !$original->needsReencryption($this->encryptor->getIdentifier(), $this->serializer->getIdentifier())
```

Comment:
Is the identifier really sufficient to determine that? What about algorithms that don't change, but are parametrized?